fineract-core/src/main/java/org/apache/fineract/commands/service/CommandSourceService.java

```diff
@@ -64,8 +66,12 @@ public CommandSource saveInitialSameTransaction(CommandWrapper wrapper, JsonComm
 
     @NotNull
     private CommandSource saveInitial(CommandWrapper wrapper, JsonCommand jsonCommand, AppUser maker, String idempotencyKey) {
-        CommandSource initialCommandSource = getInitialCommandSource(wrapper, jsonCommand, maker, idempotencyKey);
-        return commandSourceRepository.saveAndFlush(initialCommandSource);
+        try {
+            CommandSource initialCommandSource = getInitialCommandSource(wrapper, jsonCommand, maker, idempotencyKey);
+            return commandSourceRepository.saveAndFlush(initialCommandSource);
+        } catch (JpaSystemException jse) {
```

Comment:
This is not checking whether the issue was with the duplicate idempotency key.. Please de the proper checks! We want to throw IdempotentCommandProcessUnderProcessingException **only** if it failed with idempotency key duplication error!

Reply:
Done! Updated to throw the exception only when we have a SQLIntegrityConstraintViolationException